### PR TITLE
[IOTDB-5885] Add exception when measurement already exists in schema template

### PR DESCRIPTION
**Pull request by MarcosZyk**
## Description

Exception should be returned when adding an existing measurement to schema template.

```
IoTDB> show nodes in schema template t2
+----------+--------+--------+-----------+
|ChildNodes|DataType|Encoding|Compression|
+----------+--------+--------+-----------+
|        s1|   INT32|     RLE|     SNAPPY|
+----------+--------+--------+-----------+
Total line number = 1
It costs 0.027s
IoTDB> alter schema template t2 add(s1 int64)
Msg: 527: Measurement [s1] already exist in schema template t2
IoTDB> alter schema template t2 add(s2 int64)
Msg: The statement is executed successfully.
IoTDB> alter schema template t2 add(s3 int64)
Msg: The statement is executed successfully.
IoTDB> alter schema template t2 add(s1 int32, s2 int32, s3 int64)
Msg: 527: Measurement [s1, s2, s3] already exist in schema template t2

```

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/metadata/template/alter/TemplateExtendInfo.java [hunk 1]
```diff
@@ -102,6 +103,41 @@ public void addMeasurement(
     compressors.add(compressionType);
   }
 
+  public List<String> removeMeasurements(Set<String> targetMeasurementSet) {
```
Comment (Cpaulyz) [inline server/src/main/java/org/apache/iotdb/db/metadata/template/alter/TemplateExtendInfo.java:106]:
```suggestion
/**
*
* Updates this to be the difference of set between this and targetMeasurementSet. Returns the intersection.
*
* @param targetMeasurementSet The set to compare with measurement set of this.
* @return A list of elements representing the intersection between measurement set of this and the targetMeasurementSet.
*/
  public List<String> updateAsDifferenceAndGetIntersection(Set<String> targetMeasurementSet) {
```

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/schema/ClusterSchemaManager.java [hunk 2]
```diff
@@ -773,16 +773,19 @@ public synchronized TSStatus extendSchemaTemplate(TemplateExtendInfo templateExt
...
 
     Template template = resp.getTemplateList().get(0);
-    boolean needExtend = false;
-    for (String measurement : templateExtendInfo.getMeasurements()) {
-      if (!template.hasSchema(measurement)) {
-        needExtend = true;
-        break;
-      }
-    }
+    List<String> removedMeasurements =
```
Comment (Cpaulyz) [inline confignode/src/main/java/org/apache/iotdb/confignode/manager/schema/ClusterSchemaManager.java:776]:
```suggestion
    List<String> intersectionMeasurements =
```

**Review by Cpaulyz — COMMENTED** [review]
PTAL

File: server/src/main/java/org/apache/iotdb/db/metadata/template/alter/TemplateExtendInfo.java (hunk 1 above)
Comment (MarcosZyk) [reply server/src/main/java/org/apache/iotdb/db/metadata/template/alter/TemplateExtendInfo.java]: Fixed.

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/schema/ClusterSchemaManager.java (hunk 2 above)
Comment (MarcosZyk) [reply confignode/src/main/java/org/apache/iotdb/confignode/manager/schema/ClusterSchemaManager.java]: Fixed.

**Review by Cpaulyz — APPROVED** [review]
LGTM

**Pull request merged**